packages/commonwealth/client/scripts/views/pages/MyCommunityStake/Stakes/Stakes.tsx

```diff
@@ -53,11 +54,40 @@ const columnInfo = [
 const Stakes = ({ filterOptions, addressFilter }: TransactionHistoryProps) => {
   const data = useTransactionHistory({ filterOptions, addressFilter });
 
+  // aggregate transaction per community per address
+  const stakes = (() => {
+    const map = {};
+
+    data.map((x) => {
+      const key = (x.community.id + x.address).toLowerCase();
+      const action = x.action === 'mint' ? 1 : -1;
+
+      map[key] = {
+        ...x,
+        ...(map[key] || {}),
+        stake: (map[key]?.stake || 0) + x?.stake * action,
+        voteWeight: (map[key]?.voteWeight || 0) + x?.voteWeight * action,
+        avgPrice:
+          (map[key]?.avgPrice || 0) +
+          parseFloat(
+            (parseFloat(x.price) / WEI_PER_ETHER / x.stake).toFixed(5),
+          ) *
+            action,
+      };
+    });
+
+    return Object.values(map).map((x: any) => ({
+      ...x,
+      voteWeight: x.voteWeight + 1, // total vote weight is +1 of the stake weight
+      avgPrice: `${x.avgPrice.toFixed(5)} ETH`,
```

Comment:
eth is hardcoded here but in other PR we introduce other denomination like BLAST or BASE so something to keep in mind https://github.com/hicommonwealth/commonwealth/pull/7205

Reply:
fixed